service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/NotificationHandlerTest.java

```diff
@@ -67,9 +75,63 @@
     @Mock
     private GatekeeperEmailContentProvider gatekeeperEmailContentProvider;
 
+    @MockBean
+    private UserDetailsService userDetailsService;
+
     @InjectMocks
     private NotificationHandler notificationHandler;
 
+    @Test
+    void shouldNotNotifyHmctsAdminOnC2Upload() throws IOException, NotificationClientException {
+        final String SUBJ_LINE = "Lastname, SACCCCCCCC5676576567";
+        final Map<String, Object> parameters = ImmutableMap.<String, Object>builder()
+            .put("subjectLine", SUBJ_LINE)
+            .put("hearingDetailsCallout", SUBJ_LINE)
+            .build();
+
+        given(userDetailsService.getUserDetails(AUTH_TOKEN))
+            .willReturn(new UserDetails("1", "hmcts-admin@test.com",
+                "Hmcts", "Test", UserRole.HMCTS_ADMIN.getRoles()));
+
+        given(hmctsEmailContentProvider.buildC2UploadNotification(callbackRequest().getCaseDetails()))
+            .willReturn(parameters);
+
+        notificationHandler.sendNotificationForC2Upload(new C2UploadNotifyEvent(callbackRequest(), AUTH_TOKEN, USER_ID,
+            userDetailsService.getUserDetails(AUTH_TOKEN)));
+
+        verify(notificationClient, times(0)).sendEmail(
```

Comment:
I don't like this verification:

* If you remove your if statement, your code fails with NPE.
* If you fix npe by adding mocking of the court, this tests PASSES, as the values are not null.

Reply:
Hmmm... taking a look at this again. thanks for spotting

Reply:
upodated